stratum/hal/bin/barefoot/build-bf-sde.sh

```diff
@@ -139,13 +139,38 @@ export SDE_INSTALL=$SDE/install
 # Extract the SDE
 tar xf $SDE_TAR -C $SDE --strip-components 1
 
+# Get SDE version from bf-sde-[version].manifest
+SDE_VERSION=$(find "$SDE" -name 'bf-sde-*.manifest' -printf '%f')
+SDE_VERSION=${SDE_VERSION#bf-sde-} # Remove "bf-sde-"
+SDE_VERSION=${SDE_VERSION%.manifest} # Remove ".manifest"
+if [ -z "${SDE_VERSION}" ]; then
+    echo "Unknown SDE version, cannot find SDE manifest file"
+    exit 1
+else
+    echo "SDE version: ${SDE_VERSION}"
+fi
+
 # Patch stratum_profile.yaml in SDE
-cp -f $STRATUM_BF_DIR/stratum_profile.yaml $SDE/p4studio_build/profiles/
+if [[ $SDE_VERSION == "9.7.0" ]]; then
+
+    # Build BF SDE
+    pushd "$SDE/p4studio"
+    $sudo ./install-p4studio-dependencies.sh
+    ./p4studio packages extract
+    # Patch SDE to build without kernel driver
+    sed -i 's/add_subdirectory(kdrv)/#add_subdirectory(kdrv)/g' $SDE/pkgsrc/bf-drivers/CMakeLists.txt
+    ./p4studio dependencies install --source-packages bridge,libcli,thrift --jobs $JOBS
+    ./p4studio configure bfrt '^pi' '^tofino2h' '^thrift-driver' '^p4rt' tofino asic '^tofino2m' '^tofino2' '^grpc' bsp --bsp-path /bsp.tgz
+    ./p4studio build --jobs $JOBS
```

Comment:
Few reasons I use multiple commands to build SDE instead of profile (`stratum.yaml`)
1. Patch CMakeList.txt to build SDE without the `kdrv`
2. Install additional thrift dependency without enabling switched thrift feature
3. Configure with custom BSP path
4. No need to main additional file